src/lock_file/resolve/pypi.rs

```diff
@@ -344,28 +340,49 @@ pub async fn resolve_pypi(
         context.capabilities.clone(),
     );
 
-    let build_dispatch = BuildDispatch::new(
+    let build_params = UvBuildDispatchParams::new(
         &registry_client,
         &context.cache,
         Constraints::default(),
-        &interpreter,
         &index_locations,
         &flat_index,
         &dependency_metadata,
-        // TODO: could use this later to add static metadata
-        shared_state,
-        IndexStrategy::default(),
+        shared_state.clone(),
+        index_strategy,
         &config_settings,
-        build_isolation,
         LinkMode::default(),
         &build_options,
         &context.hash_strategy,
         None,
         LowerBound::default(),
         context.source_strategy,
         context.concurrency,
-    )
-    .with_build_extra_env_vars(env_variables.iter());
+    );
+
+    let interpreter = OnceCell::new();
+    let non_isolated_packages = OnceCell::new();
+    let python_env = OnceCell::new();
+
+    let pixi_build_dispatch = PixiBuildDispatch::new(
```

Comment:
Because the build dispatch can now panic when getting the interpeter, we might need to catch this and re-throw to something nice. I guess we could just wrap that around the actual resolve call, ditto for the `LookaheadSolver`.